docs/contributing/Regular-Expression-Regex.html

```diff
@@ -0,0 +1,1140 @@
+---
+layout: page
+title: Regular Expression - Regex
+---
+
+<p>The purpose of this page is to provide detailed information for using Regex while editing a masterlist, as well as searching specific terms within it. Regular expressions can formally be described as follows:</p>
+
+<blockquote cite="https://en.wikipedia.org/wiki/Regular_expression">
+A regular expression (shortened as regex or regexp) is a sequence of characters that specifies a search pattern in text.
+</blockquote>
+
+<h1>General</h1>
+<ul>
+  <li>Overview of Regular Expressions: <a href="https://www.rexegg.com/regex-quickstart.html">Quick-Start: Regex Cheat Sheet</a></li>
+  <li>Quickly test your Regex: <a href="https://regex101.com/">Regular Expressions 101</a></li>
+</ul>
+
+<h2>Advantages of Regex</h2>
+Let's take a look at <a href="https://www.nexusmods.com/skyrimspecialedition/mods/41713/">DragonPriest Retexture</a> for a first example:
+
+<p>The DragonPriest Retexture mod includes the two different plugins <code>AcolyteMasks.esl</code> and <code>AcolyteMasks.esp</code>. If we want to add a <code>url</code> for each plugin, we could write the masterlist entry like this:<p>
+
+<pre><code>
+  - name: 'AcolyteMasks.esl'
+    url:
+      - link: 'https://www.nexusmods.com/skyrimspecialedition/mods/41713/'
+        name: 'DragonPriest Retexture'
+
+  - name: 'AcolyteMasks.esp'
+    url:
+      - link: 'https://www.nexusmods.com/skyrimspecialedition/mods/41713/'
+        name: 'DragonPriest Retexture'
+</code></pre>
+
+<p>This method however is not exactly efficient, as we have added the same link twice to the masterlist. With the help of a regex we can simplify the above expression.</p>
```

Comment:
The mention of efficiency is ambiguous here: it's faster to find non-regex entries, but they take up more space. For a masterlist maintainer the main issue is maintainability, so maybe reword this to specifically mention how two entries are more of a maintenance burden than one?

Reply:
Good point, I haven't thought about that. How about this:

> <p>This writing-style is a valid method to add metadata to the masterlists and it also means that searching for the plugin entry (e.g. <code>AcolyteMasks.esl</code>) is fast and easy, as it is exactly the same as the actual plugin name.
> 
> <p>The downside to it is however, that these non-regex entries take up more space than their regex counterparts, as we have added the same link twice to the masterlist. In order to keep things maintainable, we often choose to make use of regular expressions such as the following one:</p>

I think that the word "burden" has a pretty strong negative connotation to it, as such I've tried to avoid using it.

Reply:
Yeah, that's a better description of the relevant pros and cons.

Reply:
Done in https://github.com/loot/loot.github.io/pull/109/commits/d7236457470136fd700ad24130ca86362d5e21ce,